### PR TITLE
DAOS-14322 libfabric: update to 1.19.0

### PR DESCRIPTION
**Pull request by soumagne**
Drop prov/tcp patches that were merged in 1.19.0

Drop prov/opx patch that was merged in 1.19.0

Add DmgNetworkScanTest to test-tag

### CODE REVIEW

File: libfabric.spec [hunk 1]
```diff
@@ -17,6 +17,10 @@ Release:        1%{?dist}
 %{lua:
     rpm.define("dl_version " .. string.gsub(rpm.expand("%{version}"), "~", ""))
 }
+# this is a temporary fix to work around a URL issue
+%{lua:
+    rpm.define("dl_version_fix " .. string.gsub(rpm.expand("%{version}"), "~", "."))
```
Comment (brianjmurrell) [inline libfabric.spec:22]: Oh yuck.  Are they going to fix that upstream?

File: Jenkinsfile [hunk 2]
```diff
@@ -43,4 +43,4 @@
 /* groovylint-disable-next-line CompileStatic */
 packageBuildingPipelineDAOSTest(['distros': ['el8', 'el9', 'centos7', 'leap15', 'ubuntu20.04'],
                                  'make args': 'DISTRO_VERSION_EL8=8.5',
-                                 'test-tag': 'daosio'])
+                                 'test-tag': 'daosio DmgNetworkScanTest'])
```
Comment (brianjmurrell) [inline Jenkinsfile:46]: Nice.  Glad to see people thinking about this stuff.

**Review by brianjmurrell — DISMISSED** [review]

File: libfabric.spec (hunk 1 above)
Comment (soumagne) [reply libfabric.spec]: yes this is fixed

**Review by brianjmurrell — APPROVED** [review]

**Review by frostedcmos — DISMISSED** [review]

**Review by frostedcmos — APPROVED** [review]

**Comment by soumagne** [comment]
fwiw master failed in the same test as the other mercury PR

**Comment by brianjmurrell** [comment]
If you [restart this PR](https://jenkins-1.wolf.hpdd.intel.com:9090/job/daos-stack/job/libfabric/job/PR-73/6/restart/) from the Test stage in Jenkins, it should pass now.  The [failing test](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-libfabric-PR-73-master/1/tests) is now being skipped.

It will only re-run the one failed stage and will clean up the downstream testing branches when it does pass.

**Comment by soumagne** [comment]
@brianjmurrell thanks everything passed now

**Review by brianjmurrell — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**